File: src/com/colacelli/irclib/connection/listeners/OnConnectListener.java
package com.colacelli.irclib.connection.listeners;

import com.colacelli.irclib.actors.User;
import com.colacelli.irclib.connection.Connection;
import com.colacelli.irclib.connection.Server;

public interface OnConnectListener extends Listener {
    Type TYPE = Type.CONNECT;

    void onConnect(Connection connection, Server server, User user);

    @Override
    default Type getType() {
        return TYPE;
    }
}

File: src/com/colacelli/irclib/connection/listeners/OnConnectListener.kt
package com.colacelli.irclib.connection.listeners

import com.colacelli.irclib.actors.User
import com.colacelli.irclib.connection.Connection
import com.colacelli.irclib.connection.Server

interface OnConnectListener : Listener {
    fun onConnect(connection: Connection, server: Server, user: User)

    override val type: Listener.Type
        get() = TYPE

    companion object {
        val TYPE = Listener.Type.CONNECT
    }
}
